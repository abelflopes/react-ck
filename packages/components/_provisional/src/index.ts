export * from "./image";

export * from "./avatar";

export * from "./dropdown";

export * from "./dock";

export * from "./breadcrumbs";

export * from "./scrollable-container";

export * from "./position-engine";

export * from "./top-bar";

export * from "./snackbar";

export * from "./overlay-spinner";

export * from "./menu";

export * from "./tooltip";

export * from "./popover";

export * from "./chat";

export * from "./file-uploader";

export * from "./attachment";

export * from "./snippet";
