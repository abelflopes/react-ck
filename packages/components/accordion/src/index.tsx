import React from "react";
import { List, type ListProps } from "@react-ck/list";
import { Collapse } from "@react-ck/collapse";
import styles from "./styles/index.module.scss";

export interface AccordionItem {
  header: NonNullable<React.ReactNode>;
  children: NonNullable<React.ReactNode>;
}

export interface AccordionProps extends Omit<ListProps, "items"> {
  items: AccordionItem[];
}

/**
 * An accordion is a vertically stacked list of interactive items. Each item can be "collapsed" with only a summary visible or “expanded” to show the full content for that item.
 * @param props - {@link AccordionProps}
 * @returns a React element
 */

export const Accordion = ({
  items,
  ...otherProps
}: Readonly<AccordionProps>): React.ReactElement => (
  <List
    {...otherProps}
    items={items.map(({ header, children }) => (
      <Collapse key={String(JSON.stringify(header))} header={header}>
        <div className={styles.content}>{children}</div>
      </Collapse>
    ))}
  />
);
