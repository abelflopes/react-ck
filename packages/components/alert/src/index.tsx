import React from "react";
import classNames from "classnames";
import styles from "./styles/index.module.scss";
import { Text } from "@react-ck/text";

export interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
  heading?: string;
  /** Specifies the visual style of the alert  */
  skin?: "neutral" | "primary" | "negative" | "average" | "positive";
}

// TODO: add dismiss

/**
 * Alert is a short message that provides contextual feedback in a prominent way.
 * @param props - {@link AlertProps}
 * @returns a React element
 */

export const Alert = ({
  heading,
  skin = "neutral",
  children,
  className,
  ...otherProps
}: Readonly<AlertProps>): React.ReactElement => (
  <div {...otherProps} className={classNames(styles.root, styles[skin], className)}>
    {heading && (
      <Text type="h4" as="p" variation="bold" margin={false} className={styles.heading}>
        {heading}
      </Text>
    )}
    {children}
  </div>
);
