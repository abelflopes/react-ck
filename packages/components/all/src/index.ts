export * from "./text";

export * from "./divider";

export * from "./banner";

export * from "./card";

export * from "./container";

export * from "./chip";

export * from "./collapse";

export * from "./list";

export * from "./overlay";
