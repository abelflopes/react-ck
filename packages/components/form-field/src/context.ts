import React from "react";

export interface FormFieldContextProps {
  skin: "default" | "negative" | "average" | "positive" | "ghost";
  id: string;
}

export const FormFieldContext = React.createContext<FormFieldContextProps>({
  id: "",
  skin: "default",
});

export const useFormFieldContext = (): FormFieldContextProps => React.useContext(FormFieldContext);
