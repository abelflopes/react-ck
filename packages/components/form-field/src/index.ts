export * from "./FormField";

export { useFormFieldContext } from "./context";
