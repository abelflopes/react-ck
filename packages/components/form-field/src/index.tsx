import React from "react";
import styles from "./styles/index.module.scss";
import classNames from "classnames";
import { Text } from "@react-ck/text";

export interface FormFieldProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Specifies the visual style of the form-field  */
  skin?: "default" | "negative" | "average" | "positive";
  /** The main label for the form field */
  label?: React.ReactNode;
  /** The description text for the form field */
  description?: React.ReactNode;
  /** The validation message text */
  validationMessage?: React.ReactNode;
}

/**
 * FormField is a component that provides a consistent layout and input peripherals.
 *
 * > **WARNING**: This component is used as an internal utility - if you want to render an element such as an input, use its component directly
 * @param props - {@link FormFieldProps}
 * @returns a React element
 */

export const FormField = ({
  skin = "default",
  label,
  description,
  validationMessage,
  children,
  className,
  ...otherProps
}: Readonly<FormFieldProps>): React.ReactElement => (
  <div {...otherProps} className={classNames(styles.root, styles[skin], className)}>
    {label && (
      <Text
        className={styles.label}
        variation={["bold", "small"]}
        margin={false}
        as={<label htmlFor="aaa">{label}</label>}
      />
    )}
    <div className={styles.content}>{children}</div>
    {description && (
      <Text className={styles.description} variation="small" margin={false}>
        {description}
      </Text>
    )}
    {validationMessage && (
      <Text className={styles.validation_message} variation="small" margin={false}>
        {validationMessage}
      </Text>
    )}
  </div>
);
