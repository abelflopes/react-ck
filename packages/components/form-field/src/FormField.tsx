import React, { useMemo } from "react";
import styles from "./styles/index.module.scss";
import classNames from "classnames";
import { Text } from "@react-ck/text";
import { FormFieldContext, type FormFieldContextProps } from "./context";

export interface FormFieldProps
  extends Pick<React.HTMLAttributes<HTMLDivElement>, "children" | "className"> {
  /** Specifies the visual style of the form-field  */
  skin?: FormFieldContextProps["skin"];
  /** The main label for the form field */
  label?: React.ReactNode;
  /** The description text for the form field */
  description?: React.ReactNode;
  /** The validation message text */
  validationMessage?: React.ReactNode;
}

/**
 * FormField is a component that provides a consistent layout and input peripherals.
 *
 * **WARNING**: This component is used as an internal utility - if you want to render an element such as an input, use its component directly
 * @param props - {@link FormFieldProps}
 * @returns a React element
 */

export const FormField = ({
  skin = "default",
  label,
  description,
  validationMessage,
  children,
  className,
  ...otherProps
}: Readonly<FormFieldProps>): React.ReactElement => {
  const id = React.useId();

  const context = useMemo<FormFieldContextProps>(
    () => ({
      skin,
      id,
    }),
    [id, skin],
  );

  return (
    <FormFieldContext.Provider value={context}>
      <div {...otherProps} className={classNames(styles.root, styles[skin], className)}>
        {label ? (
          <Text variation="small" margin="none" as={<label htmlFor={id}>{label}</label>} />
        ) : null}

        <div className={styles.content}>{children}</div>

        {description ? (
          <Text variation="small" margin="none">
            {description}
          </Text>
        ) : null}

        {validationMessage ? (
          <Text className={styles.validation_message} variation="small" margin="none">
            {validationMessage}
          </Text>
        ) : null}
      </div>
    </FormFieldContext.Provider>
  );
};
