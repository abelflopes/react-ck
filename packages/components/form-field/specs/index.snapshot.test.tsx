import React from "react";
import { FormField } from "../src/index";
import renderer from "react-test-renderer";

describe("Snapshot FormField", () => {
  test("renders correctly", async () => {
    const tree = renderer.create(<FormField>FormField</FormField>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
