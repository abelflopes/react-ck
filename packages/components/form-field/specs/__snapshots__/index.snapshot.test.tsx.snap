// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot FormField renders correctly 1`] = `
<div
  className="root skin_default variation_default"
>
  <div
    className="main_content"
  >
    <div
      className="input_wrapper"
    >
      FormField
    </div>
  </div>
</div>
`;
